.circleci/config.yml

```diff
@@ -0,0 +1,58 @@
+
+version: 2
+
+defaults: &defaults
+  docker:
+    - image: circleci/ruby:2.3
+  working_directory: ~/www.goodcity
+
+jobs: 
+  checkout_code:
+    <<: *defaults
+    steps:
+      - checkout
+      - save_cache:
+          key: v1-repo-{{ .Environment.CIRCLE_SHA1 }}
+          paths:
+            - ~/www.goodcity
+  test:
+    <<: *defaults
+    steps: 
+      - run: echo 'No tests to run...'
+
+  bundle_dependencies:
+    <<: *defaults
+    steps:
+      - run: gem install bundler
+      - restore_cache:
+          key: v1-repo-{{ .Environment.CIRCLE_SHA1 }}
+      - restore_cache:
+         key: v1-bundle-{{ checksum "Gemfile.lock" }}
+      - run: bundle check --path=vendor/bundle || bundle install --deployment --jobs=4 --retry=3
+      - save_cache:
+          key: v1-bundle-{{ checksum "Gemfile.lock" }}
+          paths:
+            - vendor/bundle
+  deploy:
```

Comment:
I think it might make sense to combine bundle_dependencies and deploy steps. There is no other step that requires bundle_dependencies so it feels a bit redundant to start up another docker image just to restore the cache and deploy. Much faster to combine.